### PR TITLE
feat: add spec.template.spec.securityContext 

### PR DESCRIPTION
**Pull request by ChristianBieri1995**
## 📑 Description

Adding spec.template.spec.securityContext in order to solve issues when working with Policy Management Tools such as Open Policy Agent and Gatekeeper (as in my case psp-pods-allowed-user-ranges). User and group were taken from https://github.com/k8sgpt-ai/k8sgpt/blob/main/container/Dockerfile#L37. One could also think about adding runAsNonRoot: true and fsGroup: 65532. 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

Even though being more dynamic, other prominent project such as Grafana handle the securityContext in a similar way, see https://github.com/grafana/helm-charts/blob/main/charts/grafana/values.yaml#L142-L146 & https://github.com/grafana/grafana/blob/main/Dockerfile#L106 & https://github.com/grafana/helm-charts/blob/main/charts/grafana/templates/_pod.tpl#L9-L12

### CODE REVIEW

File: charts/k8sgpt/templates/deployment.yaml [hunk 1]
```diff
@@ -21,6 +21,9 @@ spec:
         app.kubernetes.io/name: {{ include "k8sgpt.name" . }}
         app.kubernetes.io/instance: {{ .Release.Name }}
     spec:
+      securityContext:
```
[the remark this answers is not in the document]
Comment (ChristianBieri1995) [reply charts/k8sgpt/templates/deployment.yaml:24]: Sure, initially though about this one as well. Shall I change it?

**Review by AlexsJones — CHANGES_REQUESTED** [review]
How about making this configurable in the values.yaml?

File: charts/k8sgpt/templates/deployment.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (JuHyung-Son) [reply charts/k8sgpt/templates/deployment.yaml:24]: yes. make it controllerable by values

[the remark this answers is not in the document]
Comment (ChristianBieri1995) [reply charts/k8sgpt/templates/deployment.yaml:25]: Done. Did it this way as it is max. flexible (if for example fsGroup should be added in the future, only changing values.yaml is enough). Hope that's fine.

**Comment by ChristianBieri1995** [comment]
> How about making this configurable in the values.yaml?

Done, see above.

File: charts/k8sgpt/values.yaml [hunk 2]
```diff
@@ -14,7 +14,9 @@ deployment:
     requests:
       cpu: "0.2"
       memory: "156Mi"
-
+  securityContext:
```
Comment (JuHyung-Son) [inline charts/k8sgpt/values.yaml:17]: It seems better not to have a default value for securityContext.

Comment (ChristianBieri1995) [reply charts/k8sgpt/values.yaml]: I have changed to values.yaml file accordingly. No, there is a possibility to set the securityContext but it's not mandatory. Any thoughts on that?

**Review by AlexsJones — APPROVED** [review]

**Comment by AlexsJones** [comment]
@k8sgpt-ai/k8sgpt-maintainers

**Comment by arbreezy** [comment]
Should'nt we have the same logic in the [operator](https://github.com/k8sgpt-ai/k8sgpt-operator/blob/main/pkg/resources/k8sgpt.go#L105) too?

**Review by arbreezy — APPROVED** [review]

**Comment by ChristianBieri1995** [comment]
> Should'nt we have the same logic in the [operator](https://github.com/k8sgpt-ai/k8sgpt-operator/blob/main/pkg/resources/k8sgpt.go#L105) too?

Would most probably make sense ....

**Comment by AlexsJones** [comment]
Thanks for your contribution, if you have time please add to the operator @ChristianBieri1995 !

**Pull request merged**